lib/route/cls/flower.c

```diff
@@ -598,6 +649,126 @@ int rtnl_flower_get_ip_dscp(struct rtnl_cls *cls, uint8_t *dscp, uint8_t *mask)
 	return 0;
 }
 
+/**
+ * Set IPv4 source address for flower classifier
+ * @arg cls		Flower classifier.
+ * @arg addr		IPv4 source address
+ * @arg mask		mask for IPv4 source address
+ * @return 0 on success or a negative error code.
+ */
+int rtnl_flower_set_ipv4_src(struct rtnl_cls *cls, in_addr_t addr, in_addr_t mask)
+{
+	struct rtnl_flower *f;
+
+	if (!(f = rtnl_tc_data(TC_CAST(cls))))
+		return -NLE_NOMEM;
+
+	if (addr) {
+		f->cf_ipv4_src = htonl(addr);
```

Comment:
I think all those `htonl` are wrong. You copy from an `in_addr_t` to an `in_addr_t`. Neither of them is native endianness.

Reply:
Since we've spoken about this matter in other PR, there is no need to bring this here again. I will adapt functions to expect _address/mask_ in network byte order.